File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/endpoints/push/AddChannelsToPush.java
package com.pubnub.api.java.endpoints.push;

import com.pubnub.api.java.endpoints.Endpoint;
import com.pubnub.api.models.consumer.push.PNPushAddChannelResult;

public interface AddChannelsToPush extends Endpoint<PNPushAddChannelResult> {
    AddChannelsToPush pushType(com.pubnub.api.enums.PNPushType pushType);

    AddChannelsToPush channels(java.util.List<String> channels);

    AddChannelsToPush deviceId(String deviceId);

    AddChannelsToPush environment(com.pubnub.api.enums.PNPushEnvironment environment);

    AddChannelsToPush topic(String topic);
}

File: pubnub-kotlin/pubnub-kotlin-api/src/commonMain/kotlin/com/pubnub/kmp/endpoints/push/AddChannelsToPush.kt
package com.pubnub.kmp.endpoints.push

import com.pubnub.kmp.PNFuture
import com.pubnub.api.models.consumer.push.PNPushAddChannelResult

/**
 * @see [PubNub.addPushNotificationsOnChannels]
 */
interface AddChannelsToPush : PNFuture<PNPushAddChannelResult>